Pelahi/benchmarking updates

pelahi opened the pull request:
Have altered names of HIP routines and added common functions to a common interface.

pelahi committed:
Added common interface to tConvolve and updated some names in HIP, updated makefiles to use the common interface

pelahi committed:
Update to use common functions, update to names of HIP files

pelahi committed:
remove incorrectly added executable

pelahi committed:
Update to common routines

pelahi committed:
Update to common functions to include stopwatch class

pelahi committed:
Updates to codes to use common functions

pelahi committed:
Partial update to mpi to use common functions. Has Benchmark class that still requires updating

pelahi committed:
Update to OpenCL to use common functions. Note that the API for the init functions were different but it appears only opencl was different. May need further investigation to see if changes to common init functions acceptable

pelahi commented:
Don't know why CudaGridKernel.h is considered a conflicting file. Local merge seems fine

mcytowski commented:
I made some changes in CudaGridKernel.h today probably after your checkout. I have pushed those changes to repo afterwards

pelahi committed:
Merge branch 'master' into pelahi/benchmarking-updates

pelahi committed:
Updates to makefiles

Now HIP makefiles use HIP named variables for clarity. Also fixed a few bugs when trying to change the naming convention

pelahi committed:
Fixed some include statements

pelahi committed:
Removed unnecessary stopwatch files

This is now included in common.h, common.cc

pelahi committed:
Removed cub dir

pelahi committed:
HIP name fixes

Removed incorrect references to CUDA

pelahi committed:
Fixed some incorrect includes

pelahi committed:
Some code clean-up

Removed unnecessary includes, fixed some CUDA that should be HIP

pelahi committed:
Some code clean-up

Removed unnecessary includes, fixed some CUDA that should be HIP
changed names from cuda to hip

pelahi committed:
changed names from cuda to hip

pelahi committed:
Merge remote-tracking branch 'refs/remotes/origin/pelahi/benchmarking-updates' into pelahi/benchmarking-updates

pelahi committed:
Fixed incorrect cuda reference


Fixed incorrect CUDA reference

pelahi committed:
fixed incorrect cuda names

pelahi committed:
Bug fix in error message when comparison fails
